File: server/src/main/java/org/dan/jadalnia/app/festival/Fid.java
package org.dan.jadalnia.app.festival;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;
import org.dan.jadalnia.sys.type.number.ImmutableNumber;

import javax.validation.constraints.Min;

public class Fid extends ImmutableNumber {
    public static final String FESTIVAL_ID_SHOULD_BE_A_POSITIVE_NUMBER = "festival id should be a positive number";

    @JsonCreator
    public Fid(int value) {
        super(value);
    }

    @Min(value = 1, message = FESTIVAL_ID_SHOULD_BE_A_POSITIVE_NUMBER)
    public int getValidateValue() {
        return super.getValidateValue();
    }

    // jax-rsp
    @JsonCreator
    public static Fid valueOf(String s) {
        return new Fid(Integer.valueOf(s));
    }

    public static Fid of(int id) {
        return new Fid(id);
    }

    @JsonIgnore
    public int getTid() {
        return intValue();
    }
}

File: kot/src/main/kotlin/org/dan/jadalnia/app/festival/pojo/Fid.kt
package org.dan.jadalnia.app.festival.pojo;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnore;

import org.dan.jadalnia.sys.type.number.ImmutableNumber;

import javax.validation.constraints.Min;

class Fid(v: Int): ImmutableNumber(v) {
    companion object {
        const val FESTIVAL_ID_SHOULD_BE_A_POSITIVE_NUMBER
                = "festival id should be a positive number"

        @JvmStatic
        @JsonCreator // jax-rsp
        fun valueOf(s: String): Fid = Fid(Integer.valueOf(s))
        @JvmStatic
        @JsonCreator
        fun of(id: Int): Fid = Fid(id)
    }

    @JsonIgnore
    @Min(value = 1, message = FESTIVAL_ID_SHOULD_BE_A_POSITIVE_NUMBER)
    override fun getValidateValue(): Int = super.getValidateValue();
}
